src/main/java/org/sigmah/shared/dto/element/FlexibleElementDTO.java

```diff
@@ -297,6 +299,11 @@ public void componentSelected(ButtonEvent ce) {
 						loadAndShowHistory(wrapper.getHistoryButton());
 					}
 				});
+				
+				if (ElementTypeEnum.CONTACT_LIST == this.getElementType() && Profiler.INSTANCE.getApplicationStateManager().getLastState() == ApplicationState.OFFLINE) {
```

Comment:
This piece of code

```java
Profiler.INSTANCE.getApplicationStateManager().getLastState() == ApplicationState.OFFLINE
```

is copy-pasted everywhere (sometimes in a method, sometimes not). Can you factorize in a single method somewhere in `Profiler.INSTANCE` or in `ApplicationStateManager`? I guess what you are actually looking for is the flag `ApplicationStateManager#online`